src/python/pants/backend/jvm/tasks/coverage/jacoco.py

```diff
@@ -61,13 +100,69 @@ def classpath_prepend(self):
 
   @property
   def extra_jvm_options(self):
-    # TODO(jtrobec): implement code coverage using jacoco
-    return []
+    agent_option = '-javaagent:{agent}=destfile={destfile}'.format(agent=self._agent_path,
+                                                                   destfile=self._coverage_datafile)
+    return [agent_option]
+
+  @staticmethod
+  def is_coverage_target(tgt):
+    return (tgt.is_java or tgt.is_scala) and not tgt.is_test and not tgt.is_synthetic
 
   def report(self, execution_failed_exception=None):
-    # TODO(jtrobec): implement code coverage using jacoco
-    pass
+    if execution_failed_exception:
+      self._context.log.warn('Test failed: {0}'.format(execution_failed_exception))
+      if self._coverage_force:
+        self._context.log.warn('Generating report even though tests failed.')
+      else:
+        return
+
+    report_dir = os.path.join(self._settings.coverage_dir, 'reports')
+    safe_mkdir(report_dir, clean=True)
+    for report_format in ['xml', 'csv', 'html']:
```

Comment:
Report format seems like it should be a Subsystem option with a default (or perhaps a `list_option` that you can specify multiple values for?) People aren't going to need all of these.

Reply:
this is consistent with how cobertura currently functions, but i'm open to changing it. what do you think the interaction should be between coverage-open which causes the report to open after execution, and the report format? should open force html and fail if another format is specified? should open attempt to open csv/xml reports if that's what's specified?